.github/workflows/build.yml

```diff
@@ -62,22 +58,75 @@ jobs:
       - name: Build PKI packages
         run: ./build.sh --with-timestamp --work-dir=build rpm
 
+      - name: Upload PKI packages
+        uses: actions/upload-artifact@v2
+        with:
+          name: pki-build
+          path: |
+            build/RPMS/
+            build/SRPMS/
+
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v1
 
       - name: Build runner image
         uses: docker/build-push-action@v2
         with:
           context: .
+          file: tests/ipa/Dockerfile
           build-args: |
             OS_VERSION=${{ matrix.os }}
             COPR_REPO=${{ needs.init.outputs.repo }}
           tags: pki-runner
           target: pki-runner
-          outputs: type=docker,dest=pki-${{ inputs.key_container}}-runner.tar
+          outputs: type=docker,dest=pki-runner.tar
 
       - name: Store runner image
         uses: actions/cache@v3
         with:
-          key: pki-${{ inputs.key_container }}-runner-${{ matrix.os }}-${{ github.run_id }}
-          path: pki-${{ inputs.key_container }}-runner.tar
+          key: pki-runner
+          path: pki-runner.tar
```

Comment:
Reading the documentation is not clear what happen if the key exist in the parent/default branch. I think the [documentation](https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#restrictions-for-accessing-a-cache) is not clear enough. If you accept a PR the action are performed for the main branch. In that case, if a key does not exist it will be created. When you execute again the action the key exist so you have a cache hit and it should copy to the path. The problem is repeated for the following PRs because the cache should check in the base branch.

This is how I have understood the documentation.



If you